src/layout/Summary2/SummaryComponent2/SummaryComponent2.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import { ResolveComponent } from 'src/layout/Summary2/SummaryComponent2/ComponentSummary';
+import { LayoutSetSummary } from 'src/layout/Summary2/SummaryComponent2/LayoutSetSummary';
+import { PageSummary } from 'src/layout/Summary2/SummaryComponent2/PageSummary';
+import { TaskSummaryWrapper } from 'src/layout/Summary2/SummaryComponent2/TaskSummary';
+import { TaskIdStoreProvider } from 'src/layout/Summary2/taskIdStore';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+export interface ISummaryComponent2 {
+  summaryNode: LayoutNode<'Summary2'>;
+}
+
+export function _SummaryComponent2({ summaryNode }: ISummaryComponent2) {
+  if (summaryNode.item.whatToRender.type === 'layoutSet') {
+    return (
+      <LayoutSetSummary
+        layoutSetId={summaryNode.item.whatToRender.id}
+        summaryOverrides={summaryNode.item.overWriteProperties}
+      />
+    );
+  }
+
+  if (summaryNode.item.whatToRender.type === 'page') {
+    return (
+      <PageSummary
+        pageId={summaryNode.item.whatToRender.id}
+        summaryOverrides={summaryNode.item.overWriteProperties}
+      />
+    );
+  }
+
+  if (summaryNode.item.whatToRender.type === 'component') {
+    return (
+      <ResolveComponent
+        summaryProps={summaryNode.item}
+        summaryOverrides={summaryNode.item.overWriteProperties}
+      />
+    );
+  }
+
+  if (summaryNode.item.whatToRender.type === 'task') {
+    const IDSplitted = summaryNode.item.whatToRender.id.split('>');
```

Comment:
Oh, this should probably be documented in the JsonSchema/generated types. 🙌
But I'm not sure it works entirely as expected. The way I read it, if you want to reference a component in another task, that component will be found no matter the pageId you're referencing.

Reply:
We have gone for a more intuitive way of configuring this, have a look :)